p4-16/spec/P4-16-spec.mdk

```diff
@@ -4989,14 +4985,17 @@ H h3 = { ... };  // initialize h3 with a header that is valid, field f1 0, field
 # Compile-time size determination { #sec-minsizeinbits }
 
 The method calls `minSizeInBits`, `minSizeInBytes`, `maxSizeInBits`,
-and `maxSizeInBytes` can be applied to some expressions.  Each of
-these method calls evaluate to compile-time known values that return
-the minimum size in bits required to store the expression (thus the
-result type of these methods has type `int`).  None of these methods
-evaluates the expression itself.  In consequence, the expression may
-be invalid (e.g., an out-of-bounds header stack access).
-
-The method `minSizeInBytes` always returns the result of
+and `maxSizeInBytes` can be applied to certain expressions. These
+method calls return the minimum (or maximum) size in bits required to
+store the expression. Thus, the result type of these methods has type
+`int`. Except in certain situations involving type variables,
+discussed below, these method calls produce local compile-time known
```

Comment:
Either header stack size has to be local compile-time known, or header stacks must be an exception here.

Reply:
Header stacks were an omission.